firebase-firestore/src/test/java/com/google/firebase/firestore/core/QueryTest.java

```diff
@@ -553,4 +558,67 @@ public void testMatchesAllDocuments() {
     query = baseQuery.endAt(new Bound(Collections.emptyList(), true));
     assertFalse(query.matchesAllDocuments());
   }
+
+  @Test
+  public void testCanonicalIdsAreStable() {
+    // This test aims to ensure that we do not break canonical IDs, as they are used as keys in
```

Comment:
Sorry
It's the `Target`'s canonical ID that is used in Target cache. `Query` canonical ID is only used in `SpecTestCase` and I have a TODO to remove it all together.

Reply:
Done. It makes me wish we had a test class for Target though.

Reply:
Yeah, we probably should..